pkg/scaffold/helm/chart.go

```diff
@@ -265,3 +270,116 @@ func createChartFromRemote(destDir string, opts CreateChartOptions) (*chart.Char
 
 	return createChartFromDisk(destDir, chartArchive, false)
 }
+
+func GenerateRoleRules(chart *chart.Chart) ([]rbacv1.PolicyRule, []error) {
+	helmOperatorRules := []rbacv1.PolicyRule{
+		// We need this rule so tiller can read namespaces to ensure they exist
+		{
+			APIGroups: []string{""},
+			Resources: []string{"namespaces"},
+			Verbs:     []string{"get"},
+		},
+
+		// We need this role for leader election to work
+		{
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"*"},
+		},
+	}
+	customRules, errs := parseRulesFromChart(chart)
+	customRules = append(customRules, helmOperatorRules...)
+	return customRules, errs
+}
+
+func parseRulesFromChart(chart *chart.Chart) ([]rbacv1.PolicyRule, []error) {
+	var errs []error
+	groups := map[string]map[string]struct{}{}
+	gotmplRegex := regexp.MustCompile(`{{.*}}`)
+
+	for _, tmpl := range chart.GetTemplates() {
+		data := string(tmpl.GetData())
+		name := tmpl.GetName()
+		yamlTmpls := strings.Split(data, "\n---\n")
+
+		for i, yamlTmpl := range yamlTmpls {
+			var apiVersion, kind string
+
+			clean := func(s string) string {
+				s = strings.TrimSpace(s)
+				s = strings.Trim(s, "'\"")
+				return s
+			}
+
+			// Extract the apiVersion and kind from the template
+			lines := strings.Split(yamlTmpl, "\n")
+			for _, line := range lines {
+				line = strings.TrimSpace(line)
+				if strings.HasPrefix(line, "apiVersion") {
+					apiVersion = clean(strings.TrimPrefix(line, "apiVersion:"))
+				} else if strings.HasPrefix(line, "kind:") {
+					kind = clean(strings.TrimPrefix(line, "kind:"))
+				}
+			}
+
+			// Skip templates that don't look like Kubernetes resources
+			if apiVersion == "" && kind == "" {
+				continue
+			}
+
+			// If only one of kind and apiVersion is set, we won't be able to
+			// create a valid role rule, so append an error and continue.
+			if kind == "" {
+				errs = append(errs, fmt.Errorf("could not find resource kind %s[%d]", name, i))
+				continue
+			}
+			if apiVersion == "" {
+				errs = append(errs, fmt.Errorf("could not find resource apiVersion in %s[%d]", name, i))
+				continue
+			}
+
+			// Parse the apiVersion and skip with an error if it fails.
+			gv, err := schema.ParseGroupVersion(apiVersion)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("could not parse group and version from apiVersion %s in %s[%d]: %s", apiVersion, name, i, err))
+				continue
+			}
+
+			// If the group or kind contain a go template, we won't be able to
+			// create a valid role rule, so append and error and continue.
```

Comment:
```suggestion
			// create a valid role rule, so append an error and continue.
```

Reply:
I incidentally fixed this when I was reworking this PR to use the discovery API. These issues are now directly logged.